src/main/java/groovy/ui/GroovyMain.java

```diff
@@ -289,37 +304,82 @@ boolean process(CommandLine parser) throws ParameterException, IOException {
                 main.conf.setScriptBaseClass(scriptBaseClass);
             }
 
+            final List<String> transformations = new ArrayList<>();
+            if (compileStatic) {
+                transformations.add("ast(groovy.transform.CompileStatic)");
+            }
+            if (typeChecked) {
+                transformations.add("ast(groovy.transform.TypeChecked)");
+            }
+            processConfigScriptText(buildConfigScriptText(transformations), main.conf);
+
             processConfigScripts(getConfigScripts(), main.conf);
 
             main.args = arguments;
+
             return main.run();
         }
 
         private List<String> getConfigScripts() {
             List<String> scripts = new ArrayList<String>();
+
             if (this.configscript != null) {
                 scripts.add(this.configscript);
             }
+
             String configScripts = System.getProperty("groovy.starter.configscripts", null);
+
             if (configScripts != null && !configScripts.isEmpty()) {
-                scripts.addAll(StringGroovyMethods.tokenize((CharSequence) configScripts, ','));
+                scripts.addAll(StringGroovyMethods.tokenize(configScripts, ','));
             }
+
             return scripts;
         }
     }
 
     public static void processConfigScripts(List<String> scripts, CompilerConfiguration conf) throws IOException {
         if (scripts.isEmpty()) return;
+
+        GroovyShell shell = createConfigScriptsShell(conf);
+
+        for (String script : scripts) {
+            shell.evaluate(new File(script));
+        }
+    }
+
+    public static void processConfigScriptText(final String scriptText, final CompilerConfiguration conf) {
```

Comment:
Maybe no need for the `final` keyword here. Better to keep the coding style consistent with the rest of the file.